### PR TITLE
[Doc] Migrating documentation to readthedocs

### LINKED ISSUE #212: [DOC] Support versioning of documentation
As the code base grows and fixes/updates are integrated we need a clear way of showing the differences between the release version and the latest code. Documentation versioning is not supported in `ghpages`. `readthedocs`(https://readthedocs.org/) seems like the natural option since it would require minimum changes on our side given its support for `Sphinx` and the fact that we are using the `sphinx_rtd_theme` theme.

### PR DESCRIPTION
**Pull request by jacobmontiel**
This PR is in preparation for migrating our documentation from `ghpages` to `readthedocs`.

Changes proposed in this pull request:

* `readthedocs` configuration
* Documentation updates
  * Update documentation structure
  * Include new methods from the current development effort
  * Minor documentation updates
* Fixes #212 

---

Checklist

- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] Tests are included for new functionality or updated accordingly.
- [x] Travis CI build passes with no errors.
- [x] Test Coverage is maintained (threshold is -0.2%).
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

### CODE REVIEW

File: src/skmultiflow/drift_detection/hddm_w.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 class HDDM_W(BaseDriftDetector):
     """
+    Drift Detection Method based on Hoeffding’s bounds with moving weighted average-test.
```
Comment (smastelini) [inline src/skmultiflow/drift_detection/hddm_w.py]: Should not "bounds" be singular in this case?

File: src/skmultiflow/drift_detection/hddm_w.py [hunk 2]
```diff
@@ -21,7 +23,7 @@ class HDDM_W(BaseDriftDetector):
 
     Notes
     -----
-    HDDM_W [1]_ is an Online drift detection method based on McDiarmid's bounds. HDDM_W uses
+    HDDM_W [1]_ is an online drift detection method based on McDiarmid's bounds. HDDM_W uses
```
Comment (smastelini) [inline src/skmultiflow/drift_detection/hddm_w.py]: The same here

File: src/skmultiflow/drift_detection/hddm_w.py [hunk 3]
```diff
@@ -30,7 +32,7 @@ class HDDM_W(BaseDriftDetector):
     References
     ----------
     .. [1] Frías-Blanco I, del Campo-Ávila J, Ramos-Jimenez G, et al.
-       Online and non-parametric drift detection methods based on Hoeffding’s bounds[J].
+       Online and non-parametric drift detection methods based on Hoeffding’s bounds.
```
Comment (smastelini) [inline src/skmultiflow/drift_detection/hddm_w.py]: The same here

File: src/skmultiflow/drift_detection/hddm_w.py (hunk 1 above)
Comment (jacobmontiel) [reply src/skmultiflow/drift_detection/hddm_w.py]: Singular and plural work on this case (bound-calculation is the same, bound-value**s** change with the observed data). Here we use the plural to match the paper.

File: src/skmultiflow/drift_detection/hddm_w.py (hunk 2 above)
Comment (jacobmontiel) [reply src/skmultiflow/drift_detection/hddm_w.py]: Same comment as above.

File: src/skmultiflow/drift_detection/hddm_w.py (hunk 3 above)
Comment (jacobmontiel) [reply src/skmultiflow/drift_detection/hddm_w.py]: This is the paper's title so we can not change it.

**Comment by jacobmontiel** [comment]
> Just out of curiosity, will the new `readthedocs`-based documentation be available to the older skmultiflow versions?

For v0.4.1 and older we will keep the links to the documentation for offline view. For new releases, `readthedocs` will keep online versions of older (v0.5.0+) as new releases are created.

File: src/skmultiflow/drift_detection/hddm_w.py (hunk 1 above)
Comment (smastelini) [reply src/skmultiflow/drift_detection/hddm_w.py]: I got your point. Agreed

**Comment by smastelini** [comment]
Thanks for the changes @jacobmontiel. All my questions were answered accordingly.

The changes are more than welcomed!

**Pull request merged**